uint8 left
uint8 right
bool left_dir
bool right_dir
uint8 id

